### PR TITLE
[Bugfix] PHP code incorrectly rounds valid numbers

### LINKED ISSUE #4890: PHP code incorrectly rounds valid numbers
**Describe the bug**
Using a precision of 0.05 and a max value of 6 for points for a component, the PHP code will incorrectly set it to 6.05 always.

**Expected behavior**
The component should save as 6.

**To Reproduce**
1. Edit a gradeable's rubric
2. Set the precision to 0.05
3. Set the max value for component to 6
4. Hit save

**Additional context**
The problem exists in this function in the Gradeable model:
https://github.com/Submitty/Submitty/blob/c617b754c99a279997bcef35bc2cde2a0d6249e7/site/app/models/gradeable/Gradeable.php#L1157-L1179

where given a `$this->precision = 0.05` and `$points = 6`, it returns 6.05.

### PR DESCRIPTION
**Pull request by mukul-kmr-jha**
### Please check if the PR fulfills these requirements:

* [x] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)

### What is the current behavior?
Currently there is a floating point bug inside the points calculation, this behavior can be seen by setting the precision of points as 0.05 and total point value as 6, as soon as we focuses out of the points input it gets converted to  6.05 which shouldn't be the case as 6 can be the point value using 0.05 precision. 

### What is the new behavior?
The bug is removed and it closes #4890. It improves the algorithm such that it should now better handle a number of different precisions and values. Given a small enough precision, may still have some (slight) rounding bugs, but it's not clear if we're going to hit those cases in practical usage.

### Other Information
Currently working on developing test cases to check and verify the intended behavior.

### CODE REVIEW

File: site/app/libraries/NumberUtils.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ */
+
+class NumberUtils {
+
+    /**
+     * @param $x float The number to round
+     * @param $y float The number $x to be rounded with respect to this variable $y
+     * @return float The rounded result to the nearest multiple of $y
+     */
+    public static function roundPointValue($x, $y) {
```
Comment (MasterOdin) [inline site/app/libraries/NumberUtils.php]: Let's use `$value` and `$precision` as the parameter names.

We can also add the float as direct type hints and not just in phpdoc string.

**Comment by bmcutler** [comment]
Please revise the top level post for this PR to actually describe the problem and fix.
It is not acceptable to just link to the issue.  These PR notes are part of the documentation for our release notes and should not require the reader to follow links to find the problem description.

We'll need to update the submitty.org documentation and github PR template to make these expectations clear.

File: site/app/libraries/NumberUtils.php (hunk 1 above)
Comment (mukul-kmr-jha) [reply site/app/libraries/NumberUtils.php]: @MasterOdin okay

**Comment by MasterOdin** [comment]
Thanks for fixing this! :+1:

**Pull request merged**